src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -114,6 +114,8 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             _socketType = socketType;
             _protocolType = protocolType;
 
+            Debug.Assert(addressFamily != AddressFamily.InterNetworkV6 || !DualMode);
```

Comment:
@pgavlin DualMode is actually making a WinSock call on Windows. This `Debug.Assert` might come with odd side-effects. Could we instead track a `DEBUG` only variable that checks if `DualMode` has been set to true?

A `TODO` should suffice for now with the explanation to take care of this later.